### PR TITLE
Restrict unary negation to signed integers.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/libsolidity/semanticTests/storage/packed_storage_signed.sol [hunk 1]
```diff
@@ -9,7 +9,7 @@ contract C {
         returns (uint256 x1, uint256 x2, uint256 x3, uint256 x4)
     {
         a = -2;
-        b = -uint8(a) * 2;
+        b = (0 - uint8(a)) * 2;
```
Comment (axic) [inline test/libsolidity/semanticTests/storage/packed_storage_signed.sol:12]: Is this here `0 - twos_complement(-2)` ?

Comment (chriseth) [reply test/libsolidity/semanticTests/storage/packed_storage_signed.sol:12]: `uint8(-2) == type(uint8).max + 1 - 2`

**Review by ekpyron — APPROVED** [review]

File: docs/types/value-types.rst [hunk 2]
```diff
@@ -83,8 +83,8 @@ for example ``uint256(0) - uint256(1) == 2**256 - 1``. You have to take these ov
 into account when designing safe smart contracts.
 
 The expression ``-x`` is equivalent to ``(T(0) - x)`` where
-``T`` is the type of ``x``. This means that ``-x`` will not be negative
-if the type of ``x`` is an unsigned integer type. Also, ``-x`` can be
+``T`` is the type of ``x``. It can only be applied to signed types.
+The value of ``-x`` can be
 positive if ``x`` is negative. There is another caveat also resulting
```
Comment (hrkrshnn) [inline docs/types/value-types.rst:88]: > The value of ``-x`` can be positive if ``x`` is negative. 

I'm a bit confused. Why does this have to be specified here?

**Review by hrkrshnn — APPROVED** [review]

File: docs/types/value-types.rst (hunk 2 above)
Comment (chriseth) [reply docs/types/value-types.rst:88]: For some people, this is not obvious.

Comment (ekpyron) [reply docs/types/value-types.rst:88]: I stumbled across that myself as well earlier - it does feel a bit silly :-).

**Pull request merged**